attrd/commands.c

```diff
@@ -294,6 +311,54 @@ attrd_client_update(xmlNode *xml)
 
 /*!
  * \internal
+ * \brief Respond to client clear-failure request
+ *
+ * \param[in] xml         Request XML
+ */
+void
+attrd_client_clear_failure(xmlNode *xml)
+{
+#if 0
+    /* @TODO This would be most efficient, but there is currently no way to
+     * verify that all peers support the op. If that ever changes, we could
+     * enable this code.
+     */
+    if (all_peers_support_clear_failure) {
```

Comment:
Something like that would be better than nothing. It could be tied into the peer cache management so you could clear it when a peer left. I think it would depend on the cluster whether enough messages would go around to make it helpful -- e.g. some clusters don't use any RAs that set attributes. I think the best solution would be for the election_*() routines to accept a version number (or arbitrary data) to add to the election op XML, then you'd definitely get it.